src/UserSession.php

```diff
@@ -70,6 +81,14 @@ public function acquireToken($duration = null): Token
         } catch (\RuntimeException $e) {
             // @todo This catch should be a more specific exception.
             $token = $this->signIn($duration);
+            $this->logger->info(
+                'Refreshed token with a new mpx token {token} for user {username} that expires on {date}.',
```

Comment:
It's not clear to me why an exception thrown after storing in $token the result of $this->tokenCachePool->getToken() means that a token has been refreshed with a new one. And if that exception is thrown inside getToken(), shouldn't we implement the catch and the logger dependency in TockenCachePool::getToken()?

Reply:
If that exception is thrown, then the first `$token` will never be set. The class responsibilities look like:

* `UserSession`: MPX API calls - signing in and signing out.
* `TokenCachePool`: Caching MPX tokens.

If we push fetching the new token into `TokenCachePool`, then callers won't have a choice to decide how the retry logic should work without some more significant class coupling. Also, it would mean we would have to pass the MPX client and the MPX user into the TokenCachePool's constructor, which would increase it's complexity.

Is that clearer? Anything in the code that would help explain it?